vendor/github.com/OpenBazaar/multiwallet/client/blockbook/client.go

```diff
@@ -556,6 +559,9 @@ func (i *BlockBookClient) setupListeners() error {
 		return nil
 	}
 
+	// Add stored watch addresses to listenQueue if there are any
+	i.listenQueue = append(i.listenQueue, i.listenAddrs...)
```

Comment:
Access to `listenAddrs` needs to be protected by a mutex so this read isn't in a race condition with the write in `ListenAddresses`. If you change the `listenLock` to a `RWMutex` and use them appropriately in each place, that should be sufficient.

Reply:
Ok commit has been submitted to fix this.